### PR TITLE
Mark extension as reproducible

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by fmeum** [comment]
@Wyverald

File: host/extension.bzl [hunk 1]
```diff
@@ -1,3 +1,5 @@
+load("@bazel_skylib//lib:modules.bzl", "modules")
```
Comment (Wyverald) [inline host/extension.bzl:1]: this would break WORKSPACE users. let's use `mctx.extension_metadata` directly (with `hasattr`)

Comment (fmeum) [reply host/extension.bzl:1]: But WORKSPACE users have no reason to load from `extension.bzl`, right? While we could feature detect `mctx.extension_metadata`, that would in turn break Bzlmod users on pre-Bazel 7.1.0 and thus no `reproducible` parameter.

Comment (Wyverald) [reply host/extension.bzl:1]: WORKSPACE users instantiate a `host_platform_repo` from this file (see local_config_platform.WORKSPACE in Bazel source) We could alternatively maybe separate the repo rule into a separate file, but that's neat in its own way...

Comment (fmeum) [reply host/extension.bzl]: Ah, thanks for catching that. I submitted a new version that essentially inlines what skylib is doing.

**Review by Wyverald — APPROVED** [review]

**Pull request merged**